containers/lean/Dockerfile

```diff
@@ -1,14 +1,21 @@
 # hadolint ignore=DL3006
 FROM alpine
 
-# install ruby
-RUN apk --no-cache add ruby=2.7.1-r3 ruby-dev=2.7.1-r3
+# installs ruby, then bundler via gem
+# next, it install Python3, and uses pip to install pytest
+# finally it removes the cache - all in one go to prevent another layer which we don't need
+RUN set -ex && \
+    apk --no-cache add ruby=2.7.1-r3 ruby-dev=2.7.1-r3 && \
+    gem install bundler:2.1.4 && \
+    apk add --no-cache python3=3.8.5-r0 && \
+    python3 -m ensurepip && \
+    pip3 install pytest==6.1.1 && \
+    rm -rf /var/cache/apk/*
 
-# clear after install
-RUN rm -rf /var/cache/apk/*
-
-#install bundler
-RUN gem install bundler:2.1.4
+# remove languages
+RUN set -ex && \
```

Comment:
removing these will not reduce the image size (if that was the goal). The two options are to append these del commands to the chain in line 7, or to do a multistage image instead

Reply:
Good tip! I've moved them all to the same `RUN`